airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -106,6 +106,9 @@ definitions:
         type: string
       state:
         "$ref": "#/definitions/AirbyteStateBlob"
+      namespace:
+        description: Optional Source-defined namespace. Currently only used by JDBC destinations to determine what schema to write to. Airbyte streams from the same sources should have the same namespace.
```

Comment:
> Is this always true? I'm not sure if it exists in prod, but I believe we have tests where a single source has multiple namespaces

Not necessarily - we have a few tickets planned to allow streams in the CDK to define namespaces https://github.com/airbytehq/airbyte/issues/12492. This would be used for things like bases in airtable (https://github.com/airbytehq/airbyte/issues/12491).

Reply:
> Is this always true? I'm not sure if it exists in prod, but I believe we have tests where a single source has multiple namespaces

Not necessarily - we have a few tickets planned to allow streams in the CDK to define namespaces https://github.com/airbytehq/airbyte/issues/12492. This would be used for things like bases in airtable (https://github.com/airbytehq/airbyte/issues/12491).